### PR TITLE
Show GameServer name in TestGameServerReserve flakiness fail

### PR DESCRIPTION
**Pull request by markmandel**
Extra debugging, so we can dig into what is going wrong in this test.

Work on #1276

### CODE REVIEW

File: test/e2e/gameserver_test.go [hunk 1]
```diff
@@ -491,13 +491,16 @@ func TestGameServerReserve(t *testing.T) {
...
+	assert.NoError(t, err, readyGs.ObjectMeta.Name)
+	assert.Equal(t, agonesv1.GameServerStateReserved, gs.Status.State, readyGs.ObjectMeta.Name)
 
 	// it should go back after 10 seconds
 	gs, err = framework.WaitForGameServerState(readyGs, agonesv1.GameServerStateReady, 15*time.Second)
 	assert.NoError(t, err)
 	assert.Equal(t, agonesv1.GameServerStateReady, gs.Status.State)
+
+	logrus.Info("SLEEPING")
+	time.Sleep(time.Minute)
```
Comment (roberthbailey) [inline test/e2e/gameserver_test.go]: Was the sleep for local development? I don't think we want to commit that.

File: test/e2e/gameserver_test.go [hunk 2]
```diff
@@ -491,13 +491,16 @@ func TestGameServerReserve(t *testing.T) {
 	assert.Equal(t, "ACK: RESERVE\n", reply)
 
 	gs, err = framework.WaitForGameServerState(readyGs, agonesv1.GameServerStateReserved, time.Minute)
-	assert.NoError(t, err)
-	assert.Equal(t, agonesv1.GameServerStateReserved, gs.Status.State)
+	assert.NoError(t, err, readyGs.ObjectMeta.Name)
```
Comment (roberthbailey) [inline test/e2e/gameserver_test.go]: I'm not exactly sure how this prints, but something like `fmt.Sprintf("ready game server name = %s", readyGs.ObjectMeta.Name)` might make this easier to understand in the failure output.

File: test/e2e/gameserver_test.go (hunk 1 above)
Comment (markmandel) [reply test/e2e/gameserver_test.go]: :man_facepalming:

File: test/e2e/gameserver_test.go (hunk 2 above)
Comment (markmandel) [reply test/e2e/gameserver_test.go]: Done on both!

**Pull request merged**